### PR TITLE
WIP - Add DigitalOcean-CCM support

### LINKED ISSUE #8847: Add support for DigitalOcean CCM in KKP
### What would you like to be added? / User Story
As a user, I would like to be able to use LoadBalancers natively within clusters running on DigitalOcean.
For this reason, integration with out-of-tree [digitalocean-cloud-controller-manager](https://github.com/digitalocean/digitalocean-cloud-controller-manager) is required.

### Acceptance criteria
KKP uses digitalocean's cloud controller manager to support integration with the provider.

### PR DESCRIPTION
**Pull request by shatoboar**
What this PR does / why we need it:
This PR addresses https://github.com/kubermatic/kubermatic/issues/8847 and adds support for integration with the [digitalocean-cloud-controller-manager](https://github.com/digitalocean/digitalocean-cloud-controller-manager).

Which issue(s) this PR fixes:

Fixes https://github.com/kubermatic/kubermatic/issues/8847

What type of PR is this?
/kind feature

### CODE REVIEW

File: pkg/resources/cloudcontroller/digitalocean.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+			dep.Spec.Template.Spec.AutomountServiceAccountToken = pointer.Bool(false)
+
+			dep.Spec.Template.Spec.Containers = []corev1.Container{
+				{
+					Name:    ccmContainerName,
+					Image:   data.ImageRegistry(resources.RegistryDocker) + "/oss/kubernetes/digitalocean-cloud-controller-manager:v" + DigitalOceanVersion,
+					Command: []string{"/bin/digitalocean-cloud-controller-manager"},
+					Args:    getDigitalOceanFlags(data),
+					// Env: data.toke
```
Comment (ahmedwaleedmalik) [inline pkg/resources/cloudcontroller/digitalocean.go]: Remove this

File: pkg/resources/cloudcontroller/digitalocean.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	DigitalOceanCCMDeploymentName = "digitalocean-cloud-controller-manager"
+	DigitalOceanVersion           = "0.1.39"
```
Comment (ahmedwaleedmalik) [inline pkg/resources/cloudcontroller/digitalocean.go]:
```suggestion
	DigitalOceanCCMTag           = "v0.1.39"
```

File: pkg/resources/cloudcontroller/digitalocean.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+)
+
+var (
+	digitalOceanResourceRequirements = corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: resource.MustParse("50Mi"),
+			corev1.ResourceCPU:    resource.MustParse("100m"),
+		},
+	}
+)
```
Comment (ahmedwaleedmalik) [inline pkg/resources/cloudcontroller/digitalocean.go:29]: Why do we need this?

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

File: pkg/resources/cloudcontroller/digitalocean.go (hunk 3 above)
Comment (shatoboar) [reply pkg/resources/cloudcontroller/digitalocean.go:29]: The resources were mentioned in [the DO-ccm yml](https://github.com/digitalocean/digitalocean-cloud-controller-manager/blob/8d20afd329c37057ed472db5f3dc4b5025f7eda6/releases/v0.1.39.yml#L43), I thought it was necessary

**Comment by xrstf** [comment]
So what's up with this PR? :)

**Comment by shatoboar** [comment]
This issue is currently waiting on the PR from upstream Digitalocean CCM to be merged.
https://github.com/digitalocean/digitalocean-cloud-controller-manager/pull/513

**Comment by xrstf** [comment]
Upstream PR got merged, so we can progress on this one :-))

**Comment by xrstf** [comment]
This is waiting for a new release from https://github.com/digitalocean/digitalocean-cloud-controller-manager.

**Comment by xrstf** [comment]
https://github.com/digitalocean/digitalocean-cloud-controller-manager/releases/tag/v0.1.40 has been released.

**Comment by ahmedwaleedmalik** [comment]
@shatoboar if you are pre-occupied with other tasks, can I take this one over?

**Comment by shatoboar** [comment]
@ahmedwaleedmalik yes sure, I think that would make more sense

**Comment by ahmedwaleedmalik** [comment]
Closing this in favor of https://github.com/kubermatic/kubermatic/pull/11464

**Pull request closed without merging**